File: backend/src/main/java/com/example/demo/entities/Product.java
package com.example.demo.entities;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "product")
public class Product {

    @Id @GeneratedValue
    @Column(name = "product_id")
    private int productId;

    @Column(name = "name")
    private String name;

    @Column(name = "price")
    private double price;

    @Column(name = "amount")
    private int amount;

    @ManyToMany(cascade = {CascadeType.ALL})
    @JoinTable(
            name = "Orederss_Products",
            joinColumns = { @JoinColumn(name = "product_id")},
            inverseJoinColumns = { @JoinColumn(name = "order_id") }
    )
    List<Order> orders = new ArrayList<Order>();

    public Product(){};

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getAmount() {
        return amount;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }
}

File: frontend/src/app/moduls/product.ts
export class Product {
  private  id: number;
  private name: string;
  private price: number;
  private quantity: number;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || '';
    this.name = (obj && obj.name) || '';
    this.price = (obj && obj.price) || '';
    this.quantity = (obj && obj.quantity) || '';
  }

  getName(): string {
    return this.name;
  }

  getPrice(): number {
    return this.price;
  }

  getQuantity(): number {
    return this.quantity;
  }
}


